packages/gatsby-transformer-mdx-introspection/README.md

```diff
@@ -40,31 +47,122 @@ query MyQuery {
 }
 ```
 
-Please not the lowercase `apitype` in the query - the names are normalized to lowercase by the DOM parser used.
-
-Example response:
+### Example response:
 
 ```json
 {
   "data": {
     "allComponentInMdx": {
       "nodes": [
         {
+          "component": "ApiType",
           "attributes": [
             {
-              "name": "apikey",
+              "name": "apiKey",
               "value": "test"
             },
             {
               "name": "type",
               "value": "OutOfOrderPropertiesTestType"
             }
           ],
-          "page": {
-            "fields": {
-              "slug": "/api/types/"
+          "mdx": {
+            "file": {
+              "relativePath": "api/types/index.md"
             }
           }
         },
-...
+        ...
+      ]
+    }
+  }
+}
+```
+
+## Options
+
+**`lowercaseIdentifiers`** [boolean] (optional, defaults to false)
+
+Whether to have the `component` field (and `attribute.name`) be all lower case (legacy)
+
+**`trimWhitespace`** [boolean] (optional, defaults to true)
```

Comment:
Did you have a use case for keeping the whitespace yourself? Otherwise i suggest to not have that configuration option, too.   The intent of the plugin is to make the meaning of the JSX explorable and not the syntax, so I don't think a use case that needs this fits the intent of the plugin. 

Every config option multiplies the testing and code complexity and users will bring up enough other configurability needs later on top.

Reply:
The whitespace options are interesting, just because I intended to add them as an escape hatch for potential users to opt _out_ of the very simplistic algorithm I implemented. This was mostly to make it clear that the whitespace **is not** getting trimmed/collapsed according to the [standard algorithm that browsers uses](https://medium.com/@patrickbrosset/when-does-white-space-matter-in-html-b90e8a7cdd33), which takes into account HTML semantics and CSS properties. Instead, the behavior just collapses consecutive whitespace naiively, which may produce undesired effects if he contents need to be strictly parsed (which wasn't one of my specific requirements, but doesn't seem too unlikely).

I added this because the "text"/JSX contents **are** exposed via Gatsby's node model, so users would notice that the formatting might be different. However, I know that the MDX compiler already prettifies JSX itself, so source formatting may be lost anyways. With this in mind, I'm fine with removing the option, though I would also suggest combining both of the whitespace options to add an escape hatch out of the string processing (which would be simple enough for users to re-implement if they needed to tweak the way it works).